### PR TITLE
Add Karma for testing. Add isModified. Change the interface for including in Loomio

### PR DESCRIPTION
**Pull request by robguthrie**

### CODE REVIEW

File: src/restful_client.coffee [hunk 1]
```diff
@@ -14,7 +13,14 @@ module.exports = ($http, $upload) ->
 
     buildUrl: (url, params) ->
       return url unless params?
-      url + "?" + $.param(params)
```
Comment (gdpelican) [inline src/restful_client.coffee]: What wasn't working about $.param?

File: src/base_model.coffee [hunk 2]
```diff
@@ -33,33 +34,50 @@ module.exports = () ->
...
+    transformKeys = (data, transformFn) ->
+      newData = {}
+      _.each _.keys(data), (key) ->
+        newData[transformFn(key)] = data[key]
+        return
+      newData
+
+    isModified: ->
+      if @_clonedFrom?
+        not _.every @constructor.attributeNames, (attributeName) =>
```
Comment (gdpelican) [inline src/base_model.coffee]: `_.any @constructor.attributeNames, (attr) => @[attr] != @_clonedFrom[attr]` ?

**Comment by gdpelican** [comment]
:+1: Looks goood!

File: src/restful_client.coffee (hunk 1 above)
Comment (robguthrie) [inline src/restful_client.coffee]: it was the only thing that depended on jQuery and I did not like depending on jQuery

File: src/base_model.coffee [hunk 3]
```diff
@@ -33,33 +34,67 @@ module.exports = () ->
...
+      newData = {}
+      _.each _.keys(data), (key) ->
+        newData[transformFn(key)] = data[key]
+        return
+      newData
+
+    isTimeAttribute = (attributeName) ->
+      /At$/.test(attributeName)
+
+    attributeIsModified: (attributeName) ->
```
Comment (robguthrie) [inline src/base_model.coffee]: @gdpelican updated isModified.

File: src/base_model.coffee [hunk 4]
```diff
@@ -9,6 +10,7 @@ module.exports = () ->
     @searchableFields: []
 
     constructor: (recordsInterface, data, postInitializeData = {}) ->
+      @constructor.attributeNames = []
```
Comment (robguthrie) [inline src/base_model.coffee]: This is not right. .. but I'm having problems.. If we put @attributeNames under @plural: 'undefinedPlural' then all model subclasses share the same attributeNames array and it gets very big and incorrect.

Putting it here means it's overwritten each time the constructor is run.. which is also incorrect. It's kind of like we have to put @attributeNames onto all the BaseModel subclasses. 

This also happens to be the same variable we need a solution for with Eugenes most recent master merge. 

any ideas @gdpelican ?

Comment (gdpelican) [inline src/base_model.coffee]: What about `@constructor.attributesNames = @constructor.attributeNames or []`?

Comment (gdpelican) [inline src/base_model.coffee]: / `@constructor.attributeNames = [] unless @constructor.attributeNames?`

Comment (robguthrie) [inline src/base_model.coffee]: Yea I think that will do it.

File: src/restful_client.coffee (hunk 1 above)
Comment (ahdinosaur) [inline src/restful_client.coffee]: i recommend looking to [npm](https://npmjs.org) for modules when you need a specific function like this, e.g. [`jquery-param`](https://www.npmjs.com/package/jquery-param) but even better and more node-y is [`qs`](https://www.npmjs.com/package/qs) or the built-in `querystring` (which works even using browserify).

File: package.json [hunk 5]
```diff
@@ -29,9 +30,23 @@
   },
   "homepage": "https://github.com/loomio/angular_record_store",
   "dependencies": {
-    "coffeeify": "^1.1.0"
+    "coffeeify": "^1.1.0",
+    "lodash": "^3.10.1",
+    "moment": "^2.10.6"
```
Comment (ahdinosaur) [inline package.json]: you're only using `lodash` or `moment` in the tests, as in the main code you expect them to already exists on `window`, so as they are currently used they should be `devDependencies`.

Comment (ahdinosaur) [inline package.json]: actually ignore the above pedantic comment, it's lame.

moving forward, it'd be sweet to use `lodash` and `moment` normally using `require` in the code, then if consumers want to shim in their global version of these dependencies they should use [`browserify-shim`](https://github.com/thlorenz/browserify-shim).

Comment (robguthrie) [inline package.json]: thanks keen to look into that later

**Pull request merged**